Create 2025-05-21-geospatial-techno.md (#263)
geoserver/geoserver.github.io: _posts/2025-05-21-geospatial-techno.md

---
author: Nima Ghasemloo
layout: post

> title: GeoServer Installation and Upgrade Guide

Reviewer: ```suggestion
title: GeoServer Installation and Upgrade Guide on Windows
```

| [X](https://twitter.com/geospatialtechn)
)

----

### GeoServer Installation and Upgrade Guide
In this session, we will install GeoServer on Windows using the Web Archive installation method and upgrade to a new version, while retaining existing data.

If you want to access the complete tutorial, click on the [link](https://www.youtube.com/watch?v=SpTdPIRxjU0&list=PL_ITaxp1Ob4sjk24Stboa5XbO0LGdEKbL).

[![](https://img.youtube.com/vi/SpTdPIRxjU0/0.jpg)](https://www.youtube.com/watch?v=SpTdPIRxjU0&list=PL_ITaxp1Ob4sjk24Stboa5XbO0LGdEKbL)

## Introduction
GeoServer is a versatile, Java-based application compatible with various operating systems, provided a suitable Java Virtual Machine (JVM) is available. The latest versions of GeoServer have been tested with both Oracle JRE and OpenJDK.

The GeoServer WAR file is a platform-independent web archive designed for deployment on application servers. Apache Tomcat is the recommended servlet container due to its robust integration capabilities and comprehensive documentation. This setup allows multiple web applications to run concurrently, enabling GeoServer to operate alongside other Java-based services, enhancing server versatility.

> **Note.** This guide outlines the installation of GeoServer 2.25.x using Java 17 and Apache Tomcat 9, followed by upgrade instructions. To ensure you have the latest release, please visit this [link](https://geoserver.org/download/) and avoid using older versions of GeoServer.

Reviewer 2: 2.25 is really no longer supported; I understand that you made the video previously. I am glad you included the note.

## Preparing for Installation
Before proceeding, follow the steps below:

> - Backup the existing GeoServer folder (if upgrading).  

Reviewer 2: you say folder, but you may mean data directory? Actually what do you mean?

> - Backup the existing GeoServer folder (if upgrading).  

Reviewer 2: ```suggestion
- Backup the existing GeoServer folder (if upgrading).
   
  The folder ``webapps/geoserver/data`` is the data directory containing your configuration settings you wish to preserve. 
  
  The folder ``webapps/geoserver/WEB-INF/lib`` contains the deployed GeoServer web application, along with an extensions you have manually installed.
```

To download JDK 17, navigate to [adoptium.net](https://adoptium.net) and select:  
   - **Operating System:** Windows  
   - **Architecture:** x64  
   - **Package Type:** JDK  
   - **Version:** 17-LTS  

Download the `.msi` file and run it as an administrator. During installation, accept default settings and complete the setup.

## Installing Apache Tomcat
To download and install Apache Tomcat software, navigate to [tomcat.apache.org](https://tomcat.apache.org) and select **Tomcat 9** from the **Download** section.

Choose the **32-bit/64-bit Windows Service Installer** and run it as an administrator.

During setup:  
   - Configure the **ports** (default recommended).  
   - Set a **secure** username and password for administration (avoiding common defaults like `admin` or `tomcat`).  
   - The installer should auto-detect the installed JDK; if not, the user manually selects the Java installation path.   

To configure JVM memory allocation, navigate to `C:\Program Files\Apache Software Foundation\Tomcat 9.0\bin` and run **Tomcat9w.exe** as an administrator.

In the **Java** tab, the user sets:  
   - **Initial Memory Pool:** 512 MB
   - **Maximum Memory Pool:** 1024 MB
   

> Switch to the **General** tab, and set **Startup Type** to **Automatic**, and start the Tomcat service.

Reviewer 2: There is a number of other settings to do here please:

- https://docs.geoserver.org/latest/en/user/production/java.html#running-on-java-17

```
--add-exports=java.desktop/sun.awt.image=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.text=ALL-UNNAMED --add-opens=java.desktop/java.awt.font=ALL-UNNAMED --add-opens=java.desktop/sun.awt.image=ALL-UNNAMED --add-opens=java.naming/com.sun.jndi.ldap=ALL-UNNAMED --add-opens=java.desktop/sun.java2d.pipe=ALL-UNNAMED
```

And then consider system properties for GeoServer:

* https://docs.geoserver.org/latest/en/user/production/container.html

Example:
```
-XX:SoftRefLRUPolicyMSPerMB=36000
-XX:-UsePerfData
```

Reviewer 2: I have provided a suggestion above.

## Deploying GeoServer
Download the latest **GeoServer WAR** file from [geoserver.org](https://geoserver.org).  

Extract the `.war` file and copy it to `C:\Program Files\Apache Software Foundation\Tomcat 9.0\webapps`.
  
To start GeoServer:  
   - Navigate `http://localhost:8080/manager`.  
   - Login with the Tomcat credentials.  
   - Click **Start** next to the GeoServer application.
   
The user accesses GeoServer at `http://localhost:8080/geoserver` and logs in using the default credentials:  
   - **Username:** admin  
   - **Password:** geoserver  

## Upgrading GeoServer

> Stop GeoServer via the **Tomcat Manager App**, then replace the existing `geoserver` directory in `webapps` with the backup.

Reviewer 2: ```suggestion
Stop GeoServer via the **Tomcat Manager App**, then replace the existing `webapps/geoserver/data` directory with the one from your backup.
```

> Reinstall any **compatible extensions** for the new version, and restart GeoServer and verifies functionality.  

Reviewer 2: This is incorrect the ``webapps/geoserver`` folder is the result of unpacking ``geoserver.war``. By replacing the folder you would be reverting to the previous version.

Please see the upgrade instructions: https://docs.geoserver.org/latest/en/user/installation/upgrade.html

1. backup the *data directory**, if you previously just installed you will need to copy this to an external location and use a system property or environmental variable. See https://docs.geoserver.org/latest/en/user/datadirectory/index.html

2. Adding the geoserver.war, and starting up tomcat again will "deploy" geoserver, recreating the geoserver folder and replacing what was already there. This is why it is important to take the data directory out of this location or you would lose your configuration.

Reviewer 2: I have provided a suggestion above.
